syntax = "proto3";

service ServiceRegistry {
  rpc RegisterService(ServiceInfo) returns (Empty);
  rpc DiscoverService(ServiceRequest) returns (ServiceInfo);
  rpc CheckHealth(Empty) returns (HealthStatus);
}

message ServiceInfo {
  string service_name = 1;
  string host = 2;
  string port = 3;
}

message ServiceRequest {
  string service_name = 1;
}

message Empty {}

message HealthStatus {
  bool status = 1;
}
